### PR TITLE
added proxy field to encryption to allow jws key load through proxy

### PR DESCRIPTION
**Pull request by hlahlou-pp-dev**
Related to canva ask for proxy jws key load
Screen Shot 2022-03-11 at 2 58 22 PM

### CODE REVIEW

File: src/main/java/com/hyperwallet/clientsdk/util/HyperwalletEncryption.java [hunk 1]
```diff
@@ -189,6 +192,15 @@ private JWKSet loadKeySet(String keySetLocation) throws IOException, ParseExcept
             checkKeySetLocationIsFile(keySetLocation);
             return JWKSet.load(new File(keySetLocation));
         }
+
+        if (usesProxy()) {
+            if (proxyUsername != null && proxyPassword != null) {
+                Authenticator authenticator = new MultipartRequest.DefaultPasswordAuthenticator(
```
Comment (skoong) [inline src/main/java/com/hyperwallet/clientsdk/util/HyperwalletEncryption.java:198]: Minor: We should move this class (DefaultPasswordAuthenticator) out of MultipartRequest if it's used in more than one place

File: src/main/java/com/hyperwallet/clientsdk/util/HyperwalletEncryption.java [hunk 2]
```diff
@@ -310,9 +358,31 @@ public HyperwalletEncryptionBuilder jwsExpirationMinutes(Integer jwsExpirationMi
...
+        public HyperwalletEncryptionBuilder proxyPassword(String proxyPassword) {
+            this.proxyPassword = proxyPassword;
+            return this;
+        }
+
         public HyperwalletEncryption build() {
-            return new HyperwalletEncryption(encryptionAlgorithm, signAlgorithm, encryptionMethod,
+            HyperwalletEncryption hwE = new HyperwalletEncryption(encryptionAlgorithm, signAlgorithm, encryptionMethod,
                     clientPrivateKeySetLocation, hyperwalletKeySetLocation, jwsExpirationMinutes);
+            if (proxy != null)
```
Comment (skoong) [inline src/main/java/com/hyperwallet/clientsdk/util/HyperwalletEncryption.java:379]: Minor: should be the following, make sure to run code formatting before commits (see https://engineering.paypalcorp.com/confluence/display/Payouts/IntelliJ+Setup)

```
if (proxy != null) {
                hwE.setProxy(proxy);
            }
```

**Review by skoong — COMMENTED** [review]
Overall looks good to me, just make sure to run the formatting in the files you've changed 👍 . 
https://engineering.paypalcorp.com/confluence/display/Payouts/IntelliJ+Setup

File: src/main/java/com/hyperwallet/clientsdk/util/HyperwalletEncryption.java (hunk 1 above)
Comment (hlahlou-pp-dev) [reply src/main/java/com/hyperwallet/clientsdk/util/HyperwalletEncryption.java:198]: Agreed, is there a better place we can host it?

**Pull request merged**